src/canvas.py

```diff
@@ -215,10 +215,10 @@ def set_legend(self):
                     new_lines, labels,
                     loc=self.application.plot_settings.legend_position,
                     frameon=True, reverse=True)
-            else:
-                legend = self.top_right_axis.get_legend()
-                if legend is not None:
-                    legend.remove()
+        else:
+            legend = self.top_right_axis.get_legend()
+            if legend is not None:
+                legend.remove()
```

Comment:
This change reintroduces the bug, where deleting items won't make it disappear.
```suggestion
                return
        legend = self.top_right_axis.get_legend()
        if legend is not None:
            legend.remove()
```

Reply:
Yes this suggest solution works great for me. Instead of relying on an explicit else, it just ends the function using a return when successfully executed. Which is OK.

Shortened it slightly by not explicitly assigning a separate "legend" variable.